From b6a8d60b9898c33a733f765729092172cfff89b3 Mon Sep 17 00:00:00 2001
From: Pengxuan Zheng <pzheng@codeaurora.org>
Date: Tue, 3 Jul 2018 17:48:56 -0700
Subject: [PATCH 09/17] Run header-abi-diff in advisory mode for SDLLVM

To circumvent known deficiencies in the VNDK Header Checker, force
header-abi-diff to run in advisory mode for 7 libraries which fail the check if
SDLLVM is the default target compiler.

Change-Id: If6b94176bcfbd130d58692e2d031d14838d611e9
CRs-Fixed: 2170323
Signed-off-by: Joe Maples <joe@frap129.org>
---
 cc/builder.go | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/cc/builder.go b/cc/builder.go
index de1bf1f6..8f57c39a 100644
--- a/cc/builder.go
+++ b/cc/builder.go
@@ -743,6 +743,19 @@ func SourceAbiDiff(ctx android.ModuleContext, inputDump android.Path, referenceD
 	if isVndkExt {
 		localAbiCheckAllowFlags = append(localAbiCheckAllowFlags, "-allow-extensions")
 	}
+	var sdclangAbiCheckIgnoreList = []string{
+		"libbinder",
+		"libhwbinder",
+		"libprotobuf-cpp-lite",
+		"libprotobuf-cpp-full",
+		"libunwindstack",
+		"libvixl-arm64",
+		"libvixl-arm",
+	}
+	if config.SDClang && !inList("-advice-only", localAbiCheckAllowFlags) &&
+		inList(ctx.ModuleName(), sdclangAbiCheckIgnoreList) {
+		localAbiCheckAllowFlags = append(localAbiCheckAllowFlags, "-advice-only")
+	}
 
 	ctx.Build(pctx, android.BuildParams{
 		Rule:        sAbiDiff,
-- 
2.19.1

